llvm/test/tools/sycl-post-link/device-code-split/auto-module-split-3.ll

```diff
@@ -8,7 +8,8 @@
 ; RUN: FileCheck %s -input-file=%t_0.ll --check-prefixes CHECK-TU0-IR \
 ; RUN:     --implicit-check-not TU0_kernel --implicit-check-not _Z3foov
 ; RUN: FileCheck %s -input-file=%t_1.ll --check-prefixes CHECK-TU1-IR \
-; RUN:     --implicit-check-not TU1_kernel --implicit-check-not _Z4foo2v
+; RUN:     --implicit-check-not TU1_kernel --implicit-check-not _Z4foo2v \
+; RUN:     --implicit-check-not _Z4foo3v
```

Comment:
this isnt really a question about this PR i guess but i don't understand how this test works

For the check prefix `CHECK-TU1-IR`, we have an implicit not check for `_Z4foo3v`, but then on line 29 of the test we have

`; CHECK-TU1-IR: define dso_local spir_func i32 @_Z4foo3v`

Am I missing something? How is this passing? And if it's intentional what are we testing for?

Reply:
Hi @sarnex 

As shown here - https://llvm.org/docs/CommandGuide/FileCheck.html#cmdoption-filecheck-implicit-check-not
implicit-check-not makes sure there are no additional occurrences of this _Z4foo3v between the positive checks for this definition. So, it seems ok?

Thanks

Reply:
So we are checking `_Z4foo3v` does not exist before or after the single match we check for in the `CHECK-TU1-IR` line?

Reply:
I believe it passes because `--implicit-check-not` does not mean to ensure the pattern does not appear within the file, it means that the pattern does not occur between successive positive checks. So what's being checked is that there is no use of `_Z4foo3v` between the definition of `_Z3foov` and `_Z4foo3v`. However, I feel like this check is not right - if we are testing the correctness of `referenced-indirectly`, then we want implicit not checks of all the `spir_func` functions that do not have a `i32(i32)` signature and `referenced-indirectly` - so `_Z3foov`, `_Z3barIiET_S0_`, `_Z4foo1v` and `_Z4foo2v`?

Reply:
Thanks, good catch. I've updated `check-not` patterns to properly reflect which functions we are _not_ expecting to see in which module in 12b5d4ac67eaf0747f654e3fc2260eef6ccb14d8. It should resolve this concern